### DISCUSSION TITLE
Unexpected function name

### DISCUSSION THREAD

**Discussion by nix6839** [vitest-dev/vitest#2953] [Q&A] [answered]
I want to test the function name. But when I test it, the function is appended with "2" to its name. Why is this happening?

Reproduction: https://stackblitz.com/edit/vitest-dev-vitest-j9nwcn?file=index.test.ts

Code:
```ts
// index.ts
const fn = function fn() {};

const index = {
  fn,
};

export default index;
```

```ts
// index.test.ts
import { beforeEach, expect, Mock, test, vi } from 'vitest';

let mock: Mock<[typeof import('./index')], void>;

beforeEach(async () => {
  mock = vi.fn();

  const index = await import('./index');
  mock(index);
});

test('asdf', () => {
  expect(mock.mock.calls[0][0].default.fn.name).toBe('fn');
});
```

**Comment by sheremet-va** [comment, marked as answer]
This is how esbuild works. I would suggest just importing the function and test equality instead if relying on function name
